src/compiler/moduleNameResolver.ts

```diff
@@ -2392,6 +2396,42 @@ function getVersionPathsOfPackageJsonInfo(packageJsonInfo: PackageJsonInfo, stat
     return packageJsonInfo.contents.versionPaths || undefined;
 }
 
+function getPeerDependenciesOfPackageJsonInfo(packageJsonInfo: PackageJsonInfo, state: ModuleResolutionState): string | undefined {
+    if (packageJsonInfo.contents.peerDependencies === undefined) {
+        packageJsonInfo.contents.peerDependencies = readPackageJsonPeerDependencies(packageJsonInfo, state) || false;
+    }
+    return packageJsonInfo.contents.peerDependencies || undefined;
+}
+
+function readPackageJsonPeerDependencies(packageJsonInfo: PackageJsonInfo, state: ModuleResolutionState): string | undefined {
+    const peerDependencies = readPackageJsonField(packageJsonInfo.contents.packageJsonContent, "peerDependencies", "object", state);
+    if (peerDependencies === undefined) return undefined;
+    if (state.traceEnabled) trace(state.host, Diagnostics.package_json_has_a_peerDependencies_field);
+    const packageDirectory = realPath(packageJsonInfo.packageDirectory, state.host, state.traceEnabled);
+    if (packageDirectory.indexOf(".pnpm") === -1) {

```

Comment:
> ...? I don't think we should have anything pnpm-cache-directory-name-specific; this is even configurable within `pnpm` itself using `pnpm config set store-dir D:/arbitary-path` - it could be anywhere by any name

The "store" is the global store where pnpm stores its global cache; installed projects hardlink to those files into `node_modules/.pnpm` and that can't be customized.

Reply:
I believe it actually is customizable, by the related [`virtual-store-dir`](https://github.com/pnpm/pnpm/blob/d715f87b60ce79be741435ab9d4ce551f2b097c4/config/config/src/index.ts#L265) config name.

Reply:
Ah, yeah. https://pnpm.io/npmrc#virtual-store-dir

Reply:
> Moreover, you can imagine other package managers adopting this structured symlink behavior at any time

For reference Yarn in pnpm mode uses `node_modules/.store` (symlinks) and in PnP mode `.yarn/__virtual__` (virtual file system).